ec/ecresource/deploymentresource/elasticsearch_expanders.go

```diff
@@ -135,9 +144,44 @@ func expandEsResource(raw interface{}, res *models.ElasticsearchPayload) (*model
 		}
 	}
 
+	if strategy, ok := es["strategy"]; ok {
+		if s := strategy.([]interface{}); len(s) > 0 {
+			if res.Plan.Transient == nil {
+				res.Plan.Transient = &models.TransientElasticsearchPlanConfiguration{
+					Strategy: &models.PlanStrategy{},
+				}
+			}
+			expandStrategy(s, res.Plan.Transient.Strategy)
+		}
+	}
+
 	return res, nil
 }
 
+// expandStrategy expands the Configuration Strategy.
+func expandStrategy(raw interface{}, strategy *models.PlanStrategy) *models.PlanStrategy {
+	res := strategy
+	for _, rawStrategy := range raw.([]interface{}) {
+		strategyCfg, ok := rawStrategy.(map[string]interface{})
+		if !ok {
+			continue
+		}
+		rawValue := strategyCfg["type"].(string)
+		if rawValue == autodetect {
+			strategy.Autodetect = new(models.AutodetectStrategyConfig)
+		} else if rawValue == growAndShrink {
+			strategy.GrowAndShrink = new(models.GrowShrinkStrategyConfig)
+		} else if rawValue == rollingGrowAndShrink {
+			strategy.RollingGrowAndShrink = new(models.RollingGrowShrinkStrategyConfig)
+		} else if rawValue == rollingAll {
+			strategy.Rolling = &models.RollingStrategyConfig{
+				GroupBy: "__all__",
+			}
+		}
+	}
+	return res
+}
```

Comment:
IMO we should remove the return values entirely here. If we're going to keep returning something, then we should probably use the return value [here](https://github.com/elastic/terraform-provider-ec/pull/507/files#diff-ad781c304fc98e99326fac3c762954eda56bb8f7082fe00d7522bf6e43b8570dR154)

 ```suggestion
func expandStrategy(raw interface{}, strategy *models.PlanStrategy) {
	for _, rawStrategy := range raw.([]interface{}) {
		strategyCfg, ok := rawStrategy.(map[string]interface{})
		if !ok {
			continue
		}
		rawValue := strategyCfg["type"].(string)
		if rawValue == autodetect {
			strategy.Autodetect = new(models.AutodetectStrategyConfig)
		} else if rawValue == growAndShrink {
			strategy.GrowAndShrink = new(models.GrowShrinkStrategyConfig)
		} else if rawValue == rollingGrowAndShrink {
			strategy.RollingGrowAndShrink = new(models.RollingGrowShrinkStrategyConfig)
		} else if rawValue == rollingAll {
			strategy.Rolling = &models.RollingStrategyConfig{
				GroupBy: "__all__",
			}
		}
	}
}
```